pkg/runtime/controller.go

```diff
@@ -326,31 +330,34 @@ func (c *Controller) processActions(acts []*cpb.Action, handlerConfig map[string
 			ht.Associate(ic.Handler, instName)
 
 			ti := c.templateInfo[inst.Template]
-			// grouped by adapterName.
 			vAction := actions[ti.Variety]
```

Comment:
Clarification, for my understanding.. It seems along with splitting, we are also merging across actions... 
- instances across two actions with same handler, template and variety will be collapsed into one action... 
- And instances within one action with with different templatename or different variety will be split into separate actions.

Right ? Consider adding a comment about the merging logic